### PR TITLE
Use ExecutionContext.parasitic instead of DirectExecutionContext.

### PR DESCRIPTION
**Pull request by SamirTalwar-DA**
We no longer need the latter, as of Scala 2.13.

Part of #11755.

### Pull Request Checklist

- [x] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [x] Include appropriate tests
- [x] Set a descriptive title and thorough description
- [x] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [x] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description
- [ ] If you mean to change the status of a component, please make sure you keep [the Component Status page](https://github.com/digital-asset/daml/blob/main/docs/source/support/component-statuses.rst) up to date.

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

**Review by nicu-da — APPROVED** [review]

**Comment by stefanobaghino-da** [comment]
Seems to be rather part of https://github.com/digital-asset/daml/issues/11315

File: ledger/ledger-api-client/src/main/scala/com/digitalasset/ledger/client/services/testing/time/StaticTime.scala
Comment (stefanobaghino-da) [inline ledger/ledger-api-client/src/main/scala/com/digitalasset/ledger/client/services/testing/time/StaticTime.scala:96]: Please leave this unchanged and keep the `broadcastTime` aligned.

File: ledger/participant-integration-api/src/main/scala/platform/store/appendonlydao/PaginatingAsyncStream.scala
Comment (stefanobaghino-da) [inline ledger/participant-integration-api/src/main/scala/platform/store/appendonlydao/PaginatingAsyncStream.scala:78]: > run in the same thread as the query

I would challenge this claim. I would also challenge any usage of `parasitic` as well, given that it's based on this assumption, which happens to be true most of the times, but not deterministically so. But that's definitely for another time.

File: ledger/participant-state/kvutils/tools/src/main/scala/ledger/participant/state/kvutils/tools/integritycheck/IntegrityChecker.scala
Comment (stefanobaghino-da) [inline ledger/participant-state/kvutils/tools/src/main/scala/ledger/participant/state/kvutils/tools/integritycheck/IntegrityChecker.scala:324]: Should you have used `parasitic` here?

**Review by stefanobaghino-da — APPROVED** [review]

**Review by mziolekda — CHANGES_REQUESTED** [review]
@SamirTalwar-DA 
I would like you to hold off from merging this change for a couple of days. We are in the mode where a number of changes need to be backported to 1.18. A change like this that cuts across so many classes can be very disruptive to this effort. I will accept the PR when we are out of the woods.

Thanks for your understanding

**Comment by SamirTalwar-DA** [comment]
@mziolekda: Absolutely!

@stefanobaghino-da: Thanks for catching those. I will address them when we're good to go.

File: ledger/participant-state/kvutils/tools/src/main/scala/ledger/participant/state/kvutils/tools/integritycheck/IntegrityChecker.scala
Comment (SamirTalwar-DA) [reply ledger/participant-state/kvutils/tools/src/main/scala/ledger/participant/state/kvutils/tools/integritycheck/IntegrityChecker.scala:324]: I don't see why it ever used `DirectExecutionContext` when there's a perfectly good thread pool available.

File: ledger/participant-integration-api/src/main/scala/platform/store/appendonlydao/PaginatingAsyncStream.scala
Comment (SamirTalwar-DA) [reply ledger/participant-integration-api/src/main/scala/platform/store/appendonlydao/PaginatingAsyncStream.scala:78]: Agreed, which is why I deleted it. Someone else's problem, I am afraid.

**Comment by SamirTalwar-DA** [comment]
@mziolekda, can we unblock this yet?

**Review by mziolekda — APPROVED** [review]
I have confirmed that the performance of the application is the same after the change.
Nice work, thank you Samir!

**Comment by SamirTalwar-DA** [comment]
Thanks!

**Pull request merged**
